### ISSUE TITLE
[PRE REVIEW]: UnitParser, a comprehensive unit parsing library

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#127]
**Submitting author:** @varocarbas (<a href="http://orcid.org/0000-0002-9772-9716">Alvaro Carballo Garcia</a>)
**Repository:** <a href="https://github.com/varocarbas/FlexibleParser/tree/master/all_code/UnitParser/Source/" target ="_blank">https://github.com/varocarbas/FlexibleParser/tree/master/all_code/UnitParser/Source/</a>
**Version:** 1.0.6174.20320
**Editor:** Pending
**Reviewer:** Pending

**What this issue is for**

Thanks for submitting your paper to JOSS @varocarbas. The JOSS editor (shown at the top of this issue) will work with you on this issue to find a reviewer for your submission before creating the main review issue.

**Editor instructions**

The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon. I'm here to help you with some common editorial tasks for JOSS.

For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by varocarbas** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List all of Whedon's capabilities
@whedon commands

# Assign a GitHub user as the reviewer of this submission
@whedon assign @username as reviewer

# List the GitHub usernames of the JOSS editors
@whedon list editors

# List of JOSS reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Change editorial assignment
@whedon assign @username as editor

# Set the software archive DOI at the top of the issue e.g.
@whedon set 10.0000/zenodo.00000 as archive

# Open the review issue
@whedon start review
```

:construction: **Important** :construction:

This is all quite new. Please make sure you check the top of the issue after running a @whedon command (you might also need to refresh the page to see the issue update).

**Comment by varocarbas** [comment]
@whedon set 10.5281/zenodo.192338 as archive

**Comment by whedon** [comment]
I'm sorry @varocarbas, I'm afraid I can't do that. That's something only JOSS editors are allowed to do.

**Comment by arfon** [comment]
@labarba - would you be willing to edit this submission for JOSS?

**Comment by labarba** [comment]
The repository for this submission has no License file, and no README. I don't know where to start ...

**Comment by varocarbas** [comment]
@labarba It has both, although the structure of the repository/library-vs-part might be a bit confusing. Bear in mind that UnitParser is one of the parts of FlexibleParser (-> the name of the repository), which contains the main information like [license](https://github.com/varocarbas/FlexibleParser/blob/master/LICENSE) and a generic [readme file](https://github.com/varocarbas/FlexibleParser/blob/master/README.md) which links the specific readme file of each part like the [UnitParser one](https://github.com/varocarbas/FlexibleParser/blob/master/all_readme/UnitParser.md).

To avoid misunderstandings, I created a branch including only the UnitParser code and got the DOI for it (the one I tried to tell whedon to include). This branch includes all what you need, so you should better focus on it, that is:
- [Files in Zenodo](https://zenodo.org/record/192338).
- [Associated GitHub release](https://github.com/varocarbas/FlexibleParser/releases/tag/1.0.6174.20320).

I hope that everything is clear now.

**Comment by varocarbas** [comment]
@labarba Is everything OK or do you need anything else?

**Comment by arfon** [comment]
@labarba - does @varocarbas' explanation [above](https://github.com/openjournals/joss-reviews/issues/127#issuecomment-269223883) make sense?

**Comment by labarba** [comment]
I see the license and readme files. It's licensed under CC0, which is not an approved OSI license for software.

On the other hand, I don't think I would be able to find appropriate reviewers for this software. I can't think of anyone at the moment that I could ping to review a C# library. Perhaps another editor would be better able to take this one.

**Comment by arfon** [comment]
> I see the license and readme files. It's licensed under CC0, which is not an approved OSI license for software.

Thanks for pointing this out @labarba. @varocarbas - please take a look at [this thread](https://github.com/openjournals/joss/issues/179) which describes why CC0 isn't acceptable to JOSS. Is it possible for you to change the license?

> On the other hand, I don't think I would be able to find appropriate reviewers for this software. I can't think of anyone at the moment that I could ping to review a C# library. Perhaps another editor would be better able to take this one.

No problem @labarba. @openjournals/joss-editors would anyone else be able to volunteer to edit this submission?

**Comment by varocarbas** [comment]
@arfon I don't want to get involved in any discussion about what is best/worse. This is your journal, you put the rules and end of the story. But here you have some ideas to help you or anyone else to adequately understand this issue and my position:

- I am not submitting a random piece of public domain (what the thread you are referring is about). I am submitting a piece of software which I have created in its entirety (you can confirm this issue in the copyright references on the bottom of the readme file) and released as public domain.

- If you take a look at the aforementioned copyright references (i.e., [customsolvers.com/copyright/](https://customsolvers.com/copyright) or [varocarbas.com/copyright/](http://varocarbas.com/copyright)), you would see that my whole public activity may be considered public domain since a while ago (e.g., all my public codes, my comments here or in any other site, etc.). Changing the copyright of this specific piece of software (what, on the other hand, I don't even think that would make too much sense as far as the version I am submitting has been in the public domain for some months already, what I cannot change. Thus, it would be a merely formal modification with virtually no effect other than apparently fulfilling a more or less arbitrary requirement) would be kind of a problem to me. Firstly, I would have to modify all the references where I say that "all my public activity is public domain"; and, more importantly, I would set a not-clearly-justifiable exception which I don't think that will play on my favour. Note that, when I started the development of FlexibleParser, I did think about not giving it as public domain (it is, by far, the most complex piece of software which I have ever developed as open source), an idea which I rejected precisely in order to be completely consistent with my so-far "all what you see is public domain".
In summary, the short answer to your question is: sorry, but I will not modify its copyright.

- I did read the discussion you are referring in OSI before making this submission. The main idea I got there was that ["CC0 was not explicitly rejected, but..."](https://opensource.org/faq#cc-zero); additionally, I assumed that all these problems were mainly provoked by situations where a random person (not the original author) is trying to deal with a public-domain item (created by another random person). Bear in mind that public domain is the most-damaging-for-the-author form of copyright or, in other words, the most open form of openness; that's why it wouldn't make too much sense to force an author to reduce the openness of his/her contribution in order to consider it open.
Although it seems clear that this is a problematic issue when talking generally (i.e., any public domain item independently upon who is its original author), it also seems quite clear to me what is the most logical thing to do in my specific case. In any case and as said above, I am not trying to change the rules around.
Regarding your problems to find reviewers for C#, note that this language is quite similar to Java (perhaps you are luckier on this front). Also I want to highlight that .NET is completely free since some years ago, as far as the Community version of Visual Studio (the IDE of .NET) is free. 

In any case, I don't want to unnecessarily push things. I have taken a look at the other submissions and it seems clear that you are expecting certain type of languages (perhaps you should list them expressly to avoid future problems); also UnitParser (or any other part of FlexibleParser) seems too big/complex for what is usual here (you should also perhaps include a rough estimate of code size). I certainly want to continue with this submission, but would also understand this not being exactly the place for my software. 
Please, think about all this and feel completely free to be honest with me (a "now it isn't the best time" would be fine).

**Comment by varocarbas** [comment]
@labarba Sorry for having made you wait 10 hours for my reply (read my previous comment to @arfon where I am addressing some of your concerns). Although I guess that we are in different time zones, I don't want to be shown as an egoist and disrespectful asshole who puts his own problems (and/or insecurities) before others. 

I do understand that an apology cannot fix anything, but hopefully this small gesture will somehow compensate you for what might have been interpreted as a disrespectful attitude (towards you and your work as a reviewer here). 

Lastly, I want to thank you for caring so much about me and my submission. Since the very first moment, you have been extremely helpful, honest, polite and quick at responding. You are an excellent reviewer and, presumably, also an excellent person.

**Comment by labarba** [comment]
I have not accepted the reviewer role, nor the editing role.

**Comment by varocarbas** [comment]
@labarba Sorry about this misunderstanding. I retract the last sentence of my previous comment (deleted it already).

**Comment by arfon** [comment]
Thanks for your comments @varocarbas - I'll attempt to address them here:

> - I am not submitting a random piece of public domain (what the thread you are referring is about). I am submitting a piece of software which I have created in its entirety (you can confirm this issue in the copyright references on the bottom of the readme file) and released as public domain.

I understand that this is software that you have created. The issue here is that we have a strict requirement (http://joss.theoj.org/about#author_guidelines)for all of the software reviewed here to have an [OSI approved open source license](https://opensource.org/licenses). As things currently stand CC0 is _not_ one of these licenses.

> Thus, it would be a merely formal modification with virtually no effect other than apparently fulfilling a more or less arbitrary requirement) would be kind of a problem to me. Firstly, I would have to modify all the references where I say that "all my public activity is public domain"; and, more importantly, I would set a not-clearly-justifiable exception which I don't think that will play on my favour. 

I understand this and I am sympathetic to this but as The Journal of Open Source Software is an [OSI affiliate](https://opensource.org/affiliates/about), one of the requirements for affiliates is to maintain consistency in how the term 'open source' is used.

> In summary, the short answer to your question is: sorry, but I will not modify its copyright.

Understood.

> In any case, I don't want to unnecessarily push things. I have taken a look at the other submissions and it seems clear that you are expecting certain type of languages (perhaps you should list them expressly to avoid future problems);

We don't have any strict requirements on programming languages but it's certainly true that we don't have any editors/reviewers with significant experience with `C#` at this time: https://github.com/openjournals/joss/blob/master/docs/reviewers.csv

> also UnitParser (or any other part of FlexibleParser) seems too big/complex for what is usual here (you should also perhaps include a rough estimate of code size). I certainly want to continue with this submission, but would also understand this not being exactly the place for my software.

Again, no hard rules here. We're a new journal and we're all learning as we go along here.

Aside from potential issues finding and editor and reviewer it seems like we're at a stalemate here and I humbly suggest that you withdraw this submission and we'll part ways with (I hope) no hard feelings :-)

**Comment by varocarbas** [comment]
@arfon
> Aside from potential issues finding and editor and reviewer it seems like we're at a stalemate here and I humbly suggest that you withdraw this submission and we'll part ways with (I hope) no hard feelings :-)

As said, no problem with that. I do perfectly understand your position (honestly, mostly on the language and complexity/size of the code fronts; still don't quite agree with the public domain issue) and, in any case, wouldn't ever expect my own views to prevail over the defining ones of the organisation accepting/rejecting my submission. I will withdraw my submission right away.

Let me add something about your no-hard-feelings reference ("someone" has recently said me that I explain things a lot; not sure why, but I do feel like proving its point here. Hopefully, you will not mind :)): as a sensible person, I would ever feel anything other than neutral acceptation of whatever output when dealing with respectful, honest, fair and, for work-related stuff, professional people. Even in case of having to tolerate the most pathetic sample of lack of professionalism, self-esteem, respect towards others and honesty, I would never keep "hard feelings"; perhaps a bit of pity and might eventually criticise such an attitude (directly or sarcastically), mainly to make a contribution to that person (or "person") and even to a wider community (you know? Always trying to make the world a better place by openly criticising bad-to-everyone attitudes). 

In a situation like the current one, where I have to accept a given set of rules (by liking them or not), I would ever feel anything else than acceptation. On the other hand, the aforementioned pathetic behaviour (from a professional and/or human-being points of view) might advise me to maximise my power as a citizen (for the government), a user (company), knowledgeable programmer (programming-wannabe entity), etc., but only as a way to contribute towards an eventual improvement (by helping them understand how to behave, others to know what they might be dealing with or even myself to realise that I was the one being wrong).

Short version: I have ever had what you call "hard feelings". These are the typical outputs of low-self-esteem, obsessive, coward and imposing attitudes. The kind of people taking critics as personal offences, expecting their views (= they getting what they want as they want it) to always prevail and , usually, hiding behind more or less big groups of "people"/anonymity and/or promoting nepotism/partiality/arbitrariness/unfairness.

**Comment by varocarbas** [comment]
@arfon
> Aside from potential issues finding and editor and reviewer it seems like we're at a stalemate here and I humbly suggest that you withdraw this submission and we'll part ways with (I hope) no hard feelings :-)

As said, no problem with that. I do perfectly understand your position (honestly, mostly on the language and complexity/size of the code fronts; still don't quite agree with the public domain issue) and, in any case, wouldn't ever expect my own views to prevail over the defining ones of the organisation accepting/rejecting my submission. I will withdraw my submission right away.

Let me add something about your no-hard-feelings reference ("someone" has recently said me that I explain things a lot; not sure why, but I do feel like proving its point here. Hopefully, you will not mind :)). As a sensible person, I would ever feel anything other than neutral acceptation of whatever output when dealing with respectful, honest, fair and, for work-related stuff, professional people. Even in case of having to tolerate the most pathetic sample of lack of professionalism, self-esteem, respect towards others and honesty, I would never keep "hard feelings"; perhaps a bit of pity and might eventually criticise such an attitude (directly or sarcastically), mainly to make a contribution to that person (or "person") and even to a wider community (you know? Always trying to make the world a better place by openly criticising bad-to-everyone attitudes). 

In a situation like the current one, where I have to accept a given set of rules (by liking them or not), I would ever feel anything else than acceptation. On the other hand, the aforementioned pathetic behaviour (from a professional and/or human-being points of view) might advise me to maximise my power as a citizen (for the government), a user (company), a knowledgeable programmer (programming-wannabe entity), etc. and to openly criticise them; although it would be exclusively meant to improve (by helping them understand how to behave, others to know what they might be dealing with or even myself to realise that I was the one being wrong).

Short version: I have never had what you call "hard feelings". These are the typical outputs of low-self-esteem, obsessive, coward and imposing attitudes. The kind of people taking critics as personal offences, expecting their views (= they getting what they want as they want it) to always prevail and , usually, hiding behind more or less big groups of "people"/anonymity and/or promoting nepotism/partiality/arbitrariness/unfairness.

**Comment by varocarbas** [comment]
@arfon I am not sure how to withdraw my submission. 

Additionally I found [this issue](https://github.com/openjournals/joss/issues/177), where you are precisely requesting a way to allow authors to perform such an action by themselves. As far as this issue is pretty new and it is still open, I understand that there is still no way for me to withdraw my submission by my own?

**Comment by arfon** [comment]
> Additionally, I found this issue, where you are precisely requesting a way to allow authors to perform such an action by themselves. As far as this issue is pretty new and it is still open, I understand that cannot withdraw the submission by my own (?).

Yes, that's right. I'm afraid it's not possible to do this as the author right now. I'll go ahead and do this myself now.

**Comment by varocarbas** [comment]
@arfon Thanks! Excellent work!

I want to thank you and @labarba (a Spanish-speaking colleague :)) for your professional and respectful attitude. Although the final result isn't precisely ideal, I am very happy for having found this small island of good faith and correctness, where everyone is treated with respect and values like honesty and fairness always prevail; where everyone takes full responsibility for their actions and for what their own insecurities provoke in others; a place where egoist, dishonest, unfair behaviours aren't supported, but even openly criticised. 

As a remote/internet-based worker who has tolerated lots of unfair, dishonest and low-esteem-based nonsense, even as a citizen of the world (you know? Things are lately getting a bit weird everywhere), I will always be grateful for having met you. I even expect this whole thread to help some people learn the best way to behave, treat others with respect and avoid stupid bad-faith/everyone-losing environments.

(BTW, sorry for having taking so long to write this post, but have been too busy lately. I know, I know... this is a pathetically sad excuse for a not-too-relevant-but-careless-though gesture. Hopefully, you will find a place in your heart to eventually forgive me)

**Comment by arfon** [comment]
> As a remote/internet-based worker who has tolerated lots of unfair, dishonest and low-esteem-based nonsense, even as a citizen of the world (you know? Things are lately getting a bit weird everywhere), I will always be grateful for having met you. I even expect this whole thread to help some people learn the best way to behave, treat others with respect and avoid stupid bad-faith/everyone-losing environments.

❤️

**Comment by varocarbas** [comment]
@arfon Just in case that you misunderstood this part (perhaps I was too subtle and brought my critic a bit too far though), note that I was sarcastically highlight an issue which I found particularly censurable: @labarba taking almost one month to answer my polite reply to her query (+ insisting message) and only one day after you asked her. The motivation (even trying to adequately address my concerns by bearing in mind that she doesn't seem to have too much experience in C#) is IMHO irrelevant and doesn't justify a highly disrespectful (and unprofessional, if you wish) behaviour. I also think that you taking a couple of weeks to close the issue was your way to somehow support her against my (fair) complain, what I consider equally censurable.

Anyway and completely serious now, I think that this idea of having a journal for software is quite good and I guess that you aren't that bad people within your comfort-zone. Hopefully, you will continue with your good work by helping to maximise the tremendous potential of internet for all what is related to programming and software. Also hopefully you and others will understand the point of my critic and contribute towards removing prejudice-based/arbitrarily-censoring/disrespectful (to not mention dishonest/anonymous/group-based/fanatic) attitudes from what should be objective-correctness-prone environments.

**Comment by arfon** [comment]
I'm locking this conversation as this topic has gone completely off the rails. @varocarbas - you are reading way to much into my and @labarba's behaviour.

I'm sorry that you feel that you had a poor experience here.
